src/Runner.Common/JobServer.cs

```diff
@@ -169,6 +170,10 @@ private void InitializeWebsocketClient(TimeSpan delay)
                     Trace.Info($"Creating websocket client ..." + feedStreamUrl);
                     this._websocketClient = new ClientWebSocket();
                     this._websocketClient.Options.SetRequestHeader("Authorization", $"Bearer {accessToken}");
+                    var userAgent = HostContext.UserAgents;
+                    var userAgentStrings = userAgent.Select(x => x.ToString());
+                    this._websocketClient.Options.SetRequestHeader("User-Agent", string.Join(" ", userAgentStrings));
```

Comment:
`HeaderNames.UserAgent`?

Reply:
`string.Join(" ", HostContext.UserAgents.Select(x=>x.ToString()))`?

Reply:
`HeaderNames.UserAgent` needs https://docs.microsoft.com/en-us/dotnet/api/microsoft.net.http.headers.headernames?view=aspnetcore-6.0 we don't depend on it yet, should we add it just for this?

Reply:
>string.Join(" ", HostContext.UserAgents.Select(x=>x.ToString()))?

👍 , I also added the "prefix", I think I need suffix as well, know where it's coming from? https://github.com/actions/runner/pull/1791#discussion_r836669144